backends/bmv2/jsonconverter.cpp

```diff
@@ -803,13 +805,34 @@ JsonConverter::JsonConverter(const CompilerOptions& options) :
         headerParameter(nullptr), userMetadataParameter(nullptr), stdMetadataParameter(nullptr)
 {}
 
+void addSrcInfoData(Util::JsonObject* result,
```

Comment:
@cc10512 OK, thinking more about your suggestion here, and how precisely to do it.  The existing code in jsonconverter.cpp walks through a bunch of IR::Node structures, sometimes recursively, in a custom way for each type of Node object.  For each one of those that has a valid srcInfo field (type SourceInfo), I want not to print JSON, but to add some elements using the emplace() method to an in-memory JSON object that is being filled in, and will later be printed out after it is completely created.

You are suggesting that instead of all of these calls to addSrcInfoData() that I have created, each of them should instead be a call to a new IR::Node method that has a signature something like Node::addSourceInfoToJson(Util::JsonObject *jsonObj) that does the emplace() appropriate emplace() calls on that jsonObj?

If I do that, all of the addSrcInfoData() function calls that this patch adds would change to method calls to addSourceInfo().  Is that what you hope to see?